BuildProject/App.tsx

```diff
@@ -32,117 +30,131 @@ const App: () => Node = () => {
 
     PromotedMetrics.logMakeOfferAction(content);
-    logPassed('logMakeOfferAction');
+    recordTestPassed('logMakeOfferAction');
 
     PromotedMetrics.logAskQuestionAction(content);
-    logPassed('logAskQuestionAction');
+    recordTestPassed('logAskQuestionAction');
 
     PromotedMetrics.logAnswerQuestionAction(content);
-    logPassed('logAnswerQuestionAction');
+    recordTestPassed('logAnswerQuestionAction');
 
     PromotedMetrics.logCompleteSignInAction();
-    logPassed('logCompleteSignInAction');
+    recordTestPassed('logCompleteSignInAction');
 
     PromotedMetrics.logCompleteSignUpAction();
-    logPassed('logCompleteSignUpAction');
+    recordTestPassed('logCompleteSignUpAction');
 
     PromotedMetrics.logAction('custom');
-    logPassed('logAction');
+    recordTestPassed('logAction');
 
     PromotedMetrics.logActionWithType('custom', ActionType.Navigate);
-    logPassed('logActionWithType');
+    recordTestPassed('logActionWithType');
 
     PromotedMetrics.logActionWithContent('custom', ActionType.Share, content);
-    logPassed('logActionWithContent');
+    recordTestPassed('logActionWithContent');
+  }
+
+  const testCollectionView = (recordTestPassed) => {
+    PromotedMetrics.collectionViewDidMount('hello');
+    recordTestPassed('collectionViewDidMount');
+
+    PromotedMetrics.collectionViewDidChange([], 'hello');
+    recordTestPassed('collectionViewDidChange');
+
+    PromotedMetrics.collectionViewWillUnmount('hello');
+    recordTestPassed('collectionViewWillUnmount');
+  }
 
+  const testAncestorIds = (recordTestPassed) => {
     PromotedMetrics.getCurrentOrPendingAncestorIds();
-    logPassed('getCurrentOrPendingAncestorIds');
+    recordTestPassed('getCurrentOrPendingAncestorIds');
 
     const ancestorIds = {
       logUserId: 'batman',
       sessionId: 'gotham',
       viewId: 'joker'
     } as AncestorIds;
     PromotedMetrics.setAncestorIds(ancestorIds);
-    logPassed('setAncestorIds');
+    recordTestPassed('setAncestorIds');
+  }
+
+  const handleTestAll = () => {
+    var allMessages = '';
+    const recordTestPassed = (message) => {
+      allMessages += 'Passed: ' + message + '\n';
+    };
 
-    setText(allMessages);
-    Alert.alert('Passed', 'All logging passed.');
+    try {
+      testStartSession(recordTestPassed);
+      testLogEvents(recordTestPassed);
+      testCollectionView(recordTestPassed);
+      testAncestorIds(recordTestPassed);
+      allMessages += 'All logging passed';
+      setText(allMessages);
+    } catch (err) {
+      setText(err.message);
+    }
   };
 
   return (
     <SafeAreaView style={backgroundStyle}>
       <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
-      <ScrollView
-        contentInsetAdjustmentBehavior='automatic'
-        style={backgroundStyle}>
-        <Button
-          title='Test All'
-          onPress={handleTestAll}/>
-        <Text>{text}</Text>
-      </ScrollView>
+      <Button
```

Comment:
Ah, I see.  There isn't a way to trigger the tests onMount?  Or a delay after onMount?

Reply:
I wanted to make this a manual trigger for now, because I might add more tests (ie. scrolling impression logging) to the same app.